librz/asm/arch/evm/evmdis.c

```diff
@@ -0,0 +1,346 @@
+			rz_str_cpy(buf_asm, "SLT") break;
+		case 0x13:
+			rz_str_cpy(buf_asm, "SGT") break;
+		case 0x14:
+			rz_str_cpy(buf_asm, "EQ") break;
+		case 0x15:
+			rz_str_cpy(buf_asm, "ISZERO") break;
+		case 0x16:
+			rz_str_cpy(buf_asm, "AND") break;
+		case 0x17:
+			rz_str_cpy(buf_asm, "OR") break;
+		case 0x18:
+			rz_str_cpy(buf_asm, "XOR") break;
+		case 0x19:
+			rz_str_cpy(buf_asm, "NOT") break;
+		case 0x1a:
+			rz_str_cpy(buf_asm, "BYTE") break;
+		case 0x1b:
+			rz_str_cpy(buf_asm, "SHL") break;
+		case 0x1c:
+			rz_str_cpy(buf_asm, "SHR") break;
+		case 0x1d:
+			rz_str_cpy(buf_asm, "SAR") break;
+		case 0x20:
+			rz_str_cpy(buf_asm, "SHA3") break;
+		case 0x30:
+			rz_str_cpy(buf_asm, "ADDRESS") break;
+		case 0x31:
+			rz_str_cpy(buf_asm, "BALANCE") break;
+		case 0x32:
+			rz_str_cpy(buf_asm, "ORIGIN") break;
+		case 0x33:
+			rz_str_cpy(buf_asm, "CALLER") break;
+		case 0x34:
+			rz_str_cpy(buf_asm, "CALLVALUE") break;
+		case 0x35:
+			rz_str_cpy(buf_asm, "CALLDATALOAD") break;
+		case 0x36:
+			rz_str_cpy(buf_asm, "CALLDATASIZE") break;
+		case 0x37:
+			rz_str_cpy(buf_asm, "CALLDATACOPY") break;
+		case 0x38:
+			rz_str_cpy(buf_asm, "CODESIZE") break;
+		case 0x39:
+			rz_str_cpy(buf_asm, "CODECOPY") break;
+		case 0x3a:
+			rz_str_cpy(buf_asm, "GASPRICE") break;
+		case 0x3b:
+			rz_str_cpy(buf_asm, "EXTCODESIZE") break;
+		case 0x3c:
+			rz_str_cpy(buf_asm, "EXTCODECOPY") break;
+		case 0x3d:
+			rz_str_cpy(buf_asm, "RETURNDATASIZE") break;
+		case 0x3e:
+			rz_str_cpy(buf_asm, "RETURNDATACOPY") break;
+		case 0x3f:
+			rz_str_cpy(buf_asm, "EXTCODEHASH") break;
+		case 0x40:
+			rz_str_cpy(buf_asm, "BLOCKHASH") break;
+		case 0x41:
+			rz_str_cpy(buf_asm, "COINBASE") break;
+		case 0x42:
+			rz_str_cpy(buf_asm, "TIMESTAMP") break;
+		case 0x43:
+			rz_str_cpy(buf_asm, "NUMBER") break;
+		case 0x44:
+			rz_str_cpy(buf_asm, "DIFFICULTY") break;
+		case 0x45:
+			rz_str_cpy(buf_asm, "GASLIMIT") break;
+		case 0x46:
+			rz_str_cpy(buf_asm, "CHAINID") break;
+		case 0x47:
+			rz_str_cpy(buf_asm, "SELFBALANCE") break;
+		case 0x48:
+			rz_str_cpy(buf_asm, "BASEFEE") break;
+		case 0x50:
+			rz_str_cpy(buf_asm, "POP") break; 
+		case 0x51:
+			rz_str_cpy(buf_asm, "MLOAD") break;
+		case 0x52:
+			rz_str_cpy(buf_asm, "MSTORE") break;
+		case 0x53:
+			rz_str_cpy(buf_asm, "MSTORE8") break;
+		case 0x54:
+			rz_str_cpy(buf_asm, "SLOAD") break;
+		case 0x55:
+			rz_str_cpy(buf_asm, "SSTORE") break;
+		case 0x56:
+			rz_str_cpy(buf_asm, "JUMP") break;
+		case 0x57:
+			rz_str_cpy(buf_asm, "JUMPI") break;
+		case 0x58:
+			rz_str_cpy(buf_asm, "PC") break;
+		case 0x59:
+			rz_str_cpy(buf_asm, "MSIZE") break;
+		case 0x5a:
+			rz_str_cpy(buf_asm, "GAS") break;
+		case 0x5b:
+			rz_str_cpy(buf_asm, "JUMPDEST") break;
+		case 0x60:
+			instruction_size = 2;
+			snprintf(buf_asm, sizeof(buf_asm), "PUSH1 0x%x", buf[1]); break;
+		case 0x61:
+			instruction_size = 3;
+			snprintf(buf_asm, sizeof(buf_asm), "PUSH2 0x%x%x", buf[1], buf[2]); break;
```

Comment:
these `%x%x` are wrong. you need at least to set these as `%02x%02x` because `0a 0b` are printed as `0xab` which is wrong.
Also you never check if the `buf` length is big enough for these reads causing possibly oob reads.